### PR TITLE
Performance Tuning Guide

### PR DESCRIPTION
**Pull request by dbsid**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: grafana-performance-overview-dashboard.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
+title: Performance Overview 面板重要监控指标详解
+aliases: ['/docs-cn/dev/grafana-performance-overview-dashboard/','/docs-cn/dev/reference/key-monitoring-metrics/performance-overview-dashboard/']
```
Comment (qiancai) [inline grafana-performance-overview-dashboard.md:3]: 新增文档不需要加 aliases, aliases 是针对哪些曾经出现在文档站但现在删除了的文档，给它们在替代的文档上加的。
新增文档需要加 summary

File: grafana-performance-overview-dashboard.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+Performance Overview Dashboard 按总分结构对TiDB、TiKV、PD 的性能指标进行编排组织，包含了以下三部分内容：
+- 总的概览：数据库时间和 SQL 执行时间概览，通过颜色优化法，快速识别数据库负载特征和性能瓶颈
+- 资源负载：关键指标和资源利用率，包含数据库 QPS、应用和数据库的连接信息和请求命令类型、数据库内部 tso 和 kv 请求 OPS、TiDB 和 TiKV 的资源使用概况。
+- 自上而下的延迟分解：Query 延迟和连接空闲时间对比、Query 延迟分解、execute 阶段 tso 请求和 kv 请求的延迟、TiKV 内部写延迟的分解等。
+
+借助 Performance Overview Dashboard，TiDB 用户可以进行高效性能分析，确认用户响应时间的瓶颈是否在数据库中；如果数据库是整个系统的瓶颈，通过数据库时间概览和 SQL 延迟的分解，定位数据库内部的瓶颈点，并进行针对性的优化。使用方法和实际案例请参考 [TiDB 性能分析和优化手册](/performance/performance-tuning-guide.md) 和 [OLTP 负载优化案例](/performance/real-world-tuning-case.md)
+
+以下为 Performance Overview Dashboard 监控说明：
+
+以下为 Overview Dashboard 监控说明：
```
Comment (qiancai) [inline grafana-performance-overview-dashboard.md:21]:
```suggestion
```

File: grafana-performance-overview-dashboard.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- 99-not-in-txn：没有处于事务中，空闲连接的 P99 持续时间
+
+## Parse Duration / Compile Duration / Execute Duration
+- Parse Duration：SQL 语句解析耗时统计。
+- Compile Duration：将解析后的 SQL AST 编译成执行计划的耗时。
+- Execution Duration：执行 SQL 语句执行计划耗时。
+
+每个面板包含所有 TiDB 实例的平均值和 P99 值
+
+## Avg TiDB KV Request Duration
```
Comment (qiancai) [inline grafana-performance-overview-dashboard.md:98]: 所有子标题下都需要一个空行，请加一下，否则导致文档站构建失败。

File: grafana-performance-overview-dashboard.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+按 `Get`、`Prewrite`、 `Commit`类型统计在所有 TiDB 实例平均 KV Request 执行时间
+
+## Avg TiKV GRPC Duration
+按 `get`、`kv_prewrite`、 `kv_commit`类型统计在所有 TiKV 实例对 gRPC 请求平均的执行时间
```
Comment (qiancai) [inline grafana-performance-overview-dashboard.md:102]:
```suggestion
按 `get`、`kv_prewrite`、 `kv_commit`类型统计所有 TiKV 实例对 gRPC 请求的平均执行时间
```

File: grafana-performance-overview-dashboard.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+- database time: 每秒的总数据库时间
+- sql_type: 每种 SQL 语句每秒消耗的数据库时间
+
+## Database Time by SQL Phase
+- database time: 每秒的总数据库时间
+- get token/parse/compile/execute: 4 个 SQL 处理阶段每秒消耗的数据库时间
+
+execute 执行阶段为绿色，其他三个阶段偏红色系，如果非绿色的颜色占比明显，意味着在执行阶段之外数据库消耗了过多时间，需要进一步分析根源。
+
+## SQL Execute Time Overview
```
Comment (qiancai) [inline grafana-performance-overview-dashboard.md:33]: 可以考虑把所有 Execute Time 在界面上和文档里都改成 Execution Time 吗？ Execute 是个动词

File: performance-tuning-methods.md [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+**示例 1: 繁忙的负载**
+
+![TPC-C](/media/performance/tpcc_kv_conn.png)
+
+在此 TPC-C 负载中:
+
+- 每秒总的 KV 请求的数量为 104.2k。按请求数量排序，最高的请求类型为 `PessimisticsLock`、`Prewrite`、`Commit` 和 `BatchGet` 等。
+- 总的连接数为 810，三个 TiDB 实例的连接数大体均衡。活跃的连接数为 787.1。对比活跃连接数和总连接数，97% 的连接都是活跃的，通常意味着数据库是这个应用系统的性能瓶颈。
+
+**示例 2:  空闲的负载**
```
Comment (qiancai) [inline performance-tuning-methods.md:167]:
```suggestion
**示例 2：空闲的负载**
```

File: performance-tuning-methods.md [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+1. 通过对比 SQL 处理平均延迟和事务中 TiDB 连接的空闲时间，确定整个系统的瓶颈是否在 TiDB 中。
+2. 如果瓶颈在 TiDB 内部，根据数据库时间概览、颜色优化法、关键指标和资源利用率、自上而下的延迟分解，定位到性能瓶颈具体在整个分布式系统的哪个模块。
+
+### 整个系统的瓶颈是否在 TiDB 中
+
+- 如果事务中 TiDB 连接的平均空闲时间比 SQL 处理平均延迟高，说明应用的事务处理中，主要的延迟不在数据库中，数据库时间占用户响应时间比例小，可以确认瓶颈不在数据库中。
+
+    在这种情况下，需要关注数据库外部的组件，比如应用服务器硬件资源是否存在瓶颈，应用到数据库的网络延迟是否过高等。
+
+- 如果 SQL 平均处理比事务中 TiDB 连接的平均空闲时间高，说明事务中主要的瓶颈在 TiDB 内部，数据库时间占用户响应时间比例大。
```
Comment (qiancai) [inline performance-tuning-methods.md:25]:
```suggestion
- 如果 SQL 平均处理延迟比事务中 TiDB 连接的平均空闲时间高，说明事务中主要的瓶颈在 TiDB 内部，数据库时间占用户响应时间比例大。
```

File: performance-tuning-methods.md [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+## 基于数据库时间的性能优化方法
+
+TiDB 对 SQL 的处理路径和数据库时间进行了完善的测量和记录，方便定位数据库的性能瓶颈。即使在用户响应时间的性能数据缺失的情况下，基于 TiDB 数据库时间的相关性能指标，你也可以达到以下两个性能分析目标：
+
+1. 通过对比 SQL 处理平均延迟和事务中 TiDB 连接的空闲时间，确定整个系统的瓶颈是否在 TiDB 中。
+2. 如果瓶颈在 TiDB 内部，根据数据库时间概览、颜色优化法、关键指标和资源利用率、自上而下的延迟分解，定位到性能瓶颈具体在整个分布式系统的哪个模块。
+
+### 整个系统的瓶颈是否在 TiDB 中
+
+- 如果事务中 TiDB 连接的平均空闲时间比 SQL 处理平均延迟高，说明应用的事务处理中，主要的延迟不在数据库中，数据库时间占用户响应时间比例小，可以确认瓶颈不在数据库中。
```
Comment (qiancai) [inline performance-tuning-methods.md:21]:
```suggestion
- 如果事务中 TiDB 连接的平均空闲时间比 SQL 平均处理延迟高，说明应用的事务处理中，主要的延迟不在数据库中，数据库时间占用户响应时间比例小，可以确认瓶颈不在数据库中。
```

File: performance-tuning-methods.md [hunk 9]
```diff
@@ -0,0 +1,441 @@
...
+
+- 按 SQL 处理类型分解，判断哪种类型的 SQL 语句消耗数据库时间最多。对应的分解公式为：
+
+    `DB Time = Select Time + Insert Time + Update Time + Delete Time + Commit Time + ...`
+
+- 按 SQL 处理的 4 个步骤（即 get_token/parse/compile/execute）分解，判断哪个步骤消耗的时间最多。对应的分解公式为：
+
+    `DB Time = Get Token Time + Parse Time + Comiple Time + Execute Time`
+
+- 对于 execute 耗时，按照 TiDB 执行器本身的时间、TSO 等待时间、kv 请求时间和重试的执行时间，判断执行阶段的瓶颈。对应的分解公式为：
```
Comment (qiancai) [inline performance-tuning-methods.md:44]:
```suggestion
- 对于 execute 耗时，按照 TiDB 执行器本身的时间、TSO 等待时间、KV 请求时间和重试的执行时间，判断执行阶段的瓶颈。对应的分解公式为：
```

File: performance-tuning-methods.md [hunk 10]
```diff
@@ -0,0 +1,441 @@
...
+
+- Database Time by SQL Type：主要消耗时间的语句为 commit、update、select 和 insert 语句。
+- Database Time by SQL Phase：主要消耗时间的阶段为绿色的 execute 阶段。
+- SQL Execute Time Overview：执行阶段主要消耗时间的 KV 请求为绿色的 Prewrite 和 Commit。
+
+> **注意：**
+>
+> - KV 请求的总时间大于 execute time 为正常现象，因为 TiDB 执行器可能并发向多个 TiKV 发送 KV 请求，导致总的 KV 请求等待时间大于 execute time。TPC-C 负载中，事务提交时，TiDB 会向多个 TiKV 并行发送 Prewrite 和 Commit 请求，所以这个例子中 Prewrite、Commit 和 PessimisticsLock 请求的总时间明显大于 execute time。
+>
+> - execute time 也可能明显大于 KV 请求的总时间 加上 tso_wait 时间，这意味着 SQL 执行阶段主要时间花在 TiDB 执行器内部。两种常见的例子：1，TiDB 执行器从 TiKV 读取大量数据之后，需要在 TiDB 内部进行复杂的关联和聚合，消耗大量时间；2，应用的写语句锁冲突严重，频繁锁重试导致 `Retried execution time` 过长。
```
Comment (qiancai) [inline performance-tuning-methods.md:87]:
```suggestion
> - execute time 也可能明显大于 KV 请求的总时间加上 tso_wait 的时间，这意味着 SQL 执行阶段主要时间花在 TiDB 执行器内部。两种常见的例子：

>    - 例 1：TiDB 执行器从 TiKV 读取大量数据之后，需要在 TiDB 内部进行复杂的关联和聚合，消耗大量时间。
>    - 例 2：应用的写语句锁冲突严重，频繁锁重试导致 `Retried execution time` 过长。
```

File: grafana-performance-overview-dashboard.md (hunk 5 above)
Comment (dbsid) [reply grafana-performance-overview-dashboard.md:34]: 因为历史遗留的原因，统一使用 execute

File: performance-tuning-methods.md [hunk 11]
```diff
@@ -0,0 +1,453 @@
...
+>
+> 示例 3 select 语句需要从多个 TiKV 并行读取几千行数据，BatchGet 请求的总时间远大于执行时间。
+
+**示例 4： 锁争用负载**
+
+![OLTP](/media/performance/oltp_lock_contention_db_time.png)
+
+- Database Time by SQL Type：主要为 Update 语句。
+- Database Time by SQL Phase：主要消耗时间的阶段为绿色的 execute 阶段。
+- SQL Execute Time Overview：执行阶段主要消耗时间的 KV 请求为红色的悲观锁 PessimisticLock，并且 execute time 明显大于 KV 请求的总时间，这是因为应用的写语句锁冲突严重，频繁锁重试导致 `Retried execution time` 过长，目前 `Retried execution time` 消耗的时间，TiDB 还未进行测量。
```
Comment (qiancai) [inline performance-tuning-methods.md:118]:
```suggestion
- SQL Execute Time Overview：执行阶段主要消耗时间的 KV 请求为红色的悲观锁 PessimisticLock，execute time 明显大于 KV 请求的总时间，这是因为应用的写语句锁冲突严重，频繁锁重试导致 `Retried execution time` 过长。目前 `Retried execution time` 消耗的时间，TiDB 尚未进行测量。
```

File: performance-tuning-methods.md [hunk 12]
```diff
@@ -0,0 +1,453 @@
...
+由此可以判断，用户响应时间的瓶颈不在 TiDB 中。这个例子是在一个公有云环境下，因为应用和数据库不在同一个地区，应用和数据库之间的网络延迟高导致了超高的连接空闲时间。
+
+#### Parse、Compile 和 Execute Duration
+
+在 TiDB 中，从输入查询文本到返回结果的[典型处理流程](/sql-optimization-concepts.md)。
+
+SQL 在 TiDB 内部的处理分为四个阶段，get token、parse、compile 和 execute：
+
+- get token 阶段：通常只有几微秒的时间，可以忽略。 除非 TiDB 单个实例的连接数达到的 [token-limit
+]()https://docs.pingcap.com/tidb/dev/tidb-configuration-file 的限制，创建连接的时候被限流。
```
Comment (qiancai) [inline performance-tuning-methods.md:262]:
```suggestion
- get token 阶段：通常只有几微秒的时间，可以忽略。除非 TiDB 单个实例的连接数达到的 [token-limit](/tidb-configuration-file.md) 的限制，创建连接的时候被限流。
```

**Review by Yui-Song — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**